mvc/views/tag-helpers/authoring.rst

```diff
@@ -1,8 +1,376 @@
+
+A tag helper is any class that implements the `ITagHelper <https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers/ITagHelper.cs>`__ interface. However, when you author a tag helper, you generally derive from `TagHelper <https://github.com/aspnet/Razor/tree/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers>`__ because doing so gives you access to the methods and properties that are required. We will introduce the ``TagHelper`` methods and properties as we use them in this tutorial.
+
+#. Create a new ASP.NET MVC 6 project called **AuthoringTagHelpers**. You won't need authentication for this project.
+#. Create a folder to hold the tag helpers called *TagHelpers*. The *TagHelpers* folder is not required, but it is a reasonable convention. Now let's get started writing some simple tag helpers.
+
+Starting the email tag helper
+--------------------------------
+
+In this section we will write a tag helper that updates an email tag. For example:
+
+``<email>Support</email>``
+
+The server will use our email tag helper to convert that markup into the following:
+
+.. code-block:: HTML
+
+	<a href="mailto:Support@example.com">Support@example.com</a>
+
+That is, an anchor tag that makes this an email link. You might want to do this if you are writing a blog engine and need it to send email for marketing, support, and other contacts, all to the same domain.
+
+#. Add the following ``EmailTagHelper`` class to the *TagHelpers* folder.
+
+.. literalinclude:: authoring/sample/TagHlp/src/AuthoringTagHelpers/TagHelpers/z1EmailTagHelperCopy.cs
+   :language: c#
+   :lines: 2-16
+   
+Notes: 
+
+- The ``TargetElement`` attribute identifies "email" as the element to target for processing. If you don't specify a ``TargetElement`` attribute, ASP.NET MVC 6 uses a naming convention that targets elements of the root class name (minus the *TagHelper* portion of the class name). You could comment out this attribute (because the class is named **Email**\TagHelper), and it would target email tags. Using the attribute does make the intention explicit (that is, we are targeting **<email>** tags).
+- The ``EmailTagHelper`` class derives from `TagHelper <https://github.com/aspnet/Razor/tree/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers>`__. The ``TagHelper`` class provides the rich methods and properties we will examine in this tutorial.
+- The override ``Process`` method executes the tag helper.  The ``TagHelper`` class also provides an asynchronous version (``ProcessAsync``) with the same parameters.  
+- The context parameter to ``Process`` (and ``ProcessAsync``) contains information associated with the current HTML tag. We will use the context parameter to get the content of our target element.
+- The output parameter to ``Process`` (and ``ProcessAsync``) contains a stateful HTML element used to generate an HTML tag and content.
```

Comment:
contains a stateful HTML element representative of the original source used to generate an HTML tag and content.